src/injected/injectedScript.ts

```diff
@@ -124,21 +121,17 @@ export default class InjectedScript {
     };
 
     onRaf();
-    return result;
+    return { result, cancel: () => canceled = true };
```

Comment:
I think this is backwards - you should not return progress from the task, you should provide task with the progress. An easy check is to add logging into the picture - you need to provide the sink from the caller. Since your use cases are polling, it is natural for them to check for isCanceled or apply the same pattern you applied on the Node side.

Reply:
Done.